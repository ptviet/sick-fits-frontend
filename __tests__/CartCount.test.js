import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CartCount from '../components/CartCount';

describe('<CartCount />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CartCount count={5} />);
  });

  it('should render', () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should update via props', () => {
    wrapper.setProps({ count: 11 });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
