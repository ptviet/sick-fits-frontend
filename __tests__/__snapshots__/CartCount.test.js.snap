// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount /> should match the snapshot 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="5"
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        5
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;

exports[`<CartCount /> should update via props 1`] = `
<CartCount__AnimationStyles>
  <TransitionGroup
    childFactory={[Function]}
    component="div"
  >
    <CSSTransition
      className="count"
      classNames="count"
      key="11"
      timeout={
        Object {
          "enter": 500,
          "exit": 500,
        }
      }
      unmountOnExit={true}
    >
      <CartCount__Dot>
        11
      </CartCount__Dot>
    </CSSTransition>
  </TransitionGroup>
</CartCount__AnimationStyles>
`;
